doc/md/knowledge/postgres/partial-indexes.md

```diff
@@ -0,0 +1,306 @@
+---
+id: partial-indexes
+title: Partial Indexes in PostgreSQL
+slug: /knowledge/postgres/partial-indexes
+---
+
+### Overview of Partial Indexes
+
+#### What are Partial Indexes?
+
+With PostgreSQL we can create partial indexes to improve performance and reduce the cost, all while minimizing the amount of space used to store it on disk.
+
+#### Why do we need them?
+
+Let’s demonstrate why partial indexes are useful by contrasting them with a non-partial index. ​​If you have a lot of records in an indexed table, the number of records the index needs to track also grows. If the index grows in size, the disk space needed to store the index itself increases as well. 
+In many tables, different records are not accessed with uniform frequency. A subset of a table’s records might not be searched very frequently or not searched at all. Records take up precious space in your index whether they are queried or not. They are also updated when a new entry is added to the field.
+
+Partial indexes come into the picture to filter unsearched values and give you, as an engineer, a tool to index only what’s important.
```

Comment:
```suggestion
Partial indexes come into the picture to filter unsearched values and give you, as an engineer, a tool to index only what's important.
```
@priteshtm  please make sure to use `'` and not `’` as an apostrophe (replace in other places, please)